src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -0,0 +1,116 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { AzureWizardExecuteStep } from "vscode-azureextensionui";
+import { ProjectLanguage } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { cpUtils } from "../../../utils/cpUtils";
+import { IFunctionWizardContext } from "../IFunctionWizardContext";
+
+export class OpenAPICreateStep extends AzureWizardExecuteStep<IFunctionWizardContext> {
+    public priority: number = 220;
+
+    public async execute(wizardContext: IFunctionWizardContext): Promise<void> {
+        const uris: Uri[] = await this.askDocument();
+        const uri: Uri = uris[0];
+        const args: string[] = [];
+
+        const errorMessage: string = 'Please make sure you have Autorest installed. Visit https://aka.ms/autorest to find more information on Autorest and installation steps.';
+
+        if (!await isNPMInstalled()) {
+            throw new Error(localize("notSupported", `NPM is not installed. ${errorMessage}`));
+        } else if (!await isAutoRestInstalled()) {
+            throw new Error(localize("notSupported", `Autorest could not be detected on the system. ${errorMessage}`));
+        }
+
+        let namespace: string = '';
+        if (wizardContext.language === ProjectLanguage.CSharp || wizardContext.language === ProjectLanguage.Java) {
+            namespace = await askNamespace(wizardContext.language);
+        }
+
+        args.push(`--input-file:${cpUtils.wrapArgInQuotes(uri.fsPath)} --version:3.0.6314`);
+
+        if (wizardContext.language === ProjectLanguage.TypeScript) {
+            args.push('--azure-functions-typescript');
+            args.push('--no-namespace-folders:True');
+        } else if (wizardContext.language === ProjectLanguage.CSharp) {
+            args.push(`--namespace:${namespace}`);
+            args.push('--azure-functions-csharp');
+        } else if (wizardContext.language === ProjectLanguage.Java) {
+            args.push(`--namespace:${namespace}`);
+            args.push('--azure-functions-java');
+        } else if (wizardContext.language === ProjectLanguage.Python) {
+            args.push('--azure-functions-python');
+            args.push('--no-namespace-folders:True');
+            args.push('--no-async');
+        } else {
+            throw new Error(localize('notSupported', 'Not a supported language. We currently support C#, Java, Python, and Typescript'));
+        }
+
+        args.push('--generate-metadata:false');
+        args.push(`--output-folder:${wizardContext.projectPath}`);
+
+        ext.outputChannel.show();
```

Comment:
We try to avoid popping up the output channel because it can take a large chunk of screen real-estate and get annoying. If you really think the output is important, you can include a link in the message like [this](https://github.com/microsoft/vscode-azuretools/blob/5999c2ad4423e86f22d2c648027242d8816a50e4/appservice/src/deploy/deploy.ts#L70) to pop it up

Reply:
I've shortened it and removed extra log lines from it. I felt the pop-up is important as the OpenAPI generation can take some time and users may want to know what is happening in the background. The idea of adding the output_window link is great. I did that - but when I click the output window link now, it doesn't open the window.

I read [this](https://github.com/microsoft/vscode-azuretools/blob/5999c2ad4423e86f22d2c648027242d8816a50e4/appservice/src/tree/DeploymentTreeItem.ts#L30) comment - Do I need to register my command somewhere to allow the output_window link to be clickable?
```typescript
/**
 * NOTE: This leverages two commands prefixed with `ext.prefix` that should be registered by each extension: "showOutputChannel" and "viewDeploymentLogs"
 */
```

Reply:
You should use this command, which is already registered: https://github.com/microsoft/vscode-azurefunctions/blob/main/src/commands/registerCommands.ts#L109